Offroad-iOS/Offroad-iOS/Presentation/QuestList/Views/QuestListCollectionView.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  QuestListCollectionView.swift
+//  Offroad-iOS
+//
+//  Created by 김민성 on 4/4/25.
+//
+
+import UIKit
+
+/// 퀘스트 목록을 보여줄 collection view
+final class QuestListCollectionView: UICollectionView, ORBEmptyCaseStyle, ORBCenterLoadingStyle, ORBScrollLoadingStyle {
+    
+    typealias placeholder = QuestListEmptyPlaceholder
+    
+    // MARK: - UI Properties
+    
+    let emptyPlaceholder = QuestListEmptyPlaceholder()
+    
+    // MARK: - Life Cycle
+    
+    init() {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        layout.sectionInset = .init(top: 20, left: 24, bottom: 0, right: 24)
+        layout.minimumLineSpacing = 16
+        layout.minimumInteritemSpacing = 100
+        layout.estimatedItemSize.width = UIScreen.current.bounds.width - 32
+        
+        super.init(frame: .zero, collectionViewLayout: layout)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+}
+
+extension QuestListCollectionView {
+    
+    func showEmptyPlaceholder() {
+        showEmptyPlaceholder(view: emptyPlaceholder)
+    }
+    
+    func setEmptyState(_ emptyState: Bool) {
+        if emptyState {
```

Comment:
P5

if문과 else문에 각각 함수 한 줄만 포함되어 있기 때문에, 아래처럼 삼항 연산자를 사용하는 방식이 더 간결하고 깔끔하다고 생각하는데 어떻게 생각하시나요?
```suggestion
        func setEmptyState(_ emptyState: Bool) {
               emptyState ? showEmptyPlaceholder() : removeEmptyPlaceholder()
        }
```

Reply:
저는 여기서 삼항 연산자를 쓸까 잠깐 고민했었는데, 결과적으로는 사용하지 않는 게 더 가독성이 있다고 생각했습니다.
저는 보통 상수나 변수에 값을 할당할 때 조건에 따라 다른 값을 할당하고자 하는 경우 삼항 연산자를 사용하는데요,
이 경우에는 직관적이고 깔끔해지지만, 지금처럼 함수를 호출하는 경우에는 코드가 짧아지긴 해도, 읽기에는 직관적이지 않는 것 같더라구요.
[🔗 Swift Programming Language Guide](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/basicoperators#Ternary-Conditional-Operator) 에서 든 예시 코드도 값을 할당하는 데 쓰이고 있기도 하고, 지나치게 사용하는 경우 코드가 간결해지면서 가독성을 해칠 수 있으니 조심해서 사용하라고도 적혀있습니다.
> "Use the ternary conditional operator with care, however. Its conciseness can lead to hard-to-read code if overused."

조금 다른 이야기지만, Java나 C++ 같은 언어에서는 이러한 방법으로 삼항 연산자를 사용하면 컴파일 에러가 발생하기도 합니다. (이 언어들에서는 삼항연산자에서 비교하는 두 표현식이 반드시 어떤 '값' 을 반환하는 표현식이어야 하거든요. 지금과 같은 경우는 두 함수 모두 Void를 반환하기에 해당 언어들에서는 컴파일되지 않습니다.)

그래서 저는 조금 더 코드가 길어져도 직관적인 `if~else` 문을 사용하였습니다.

Reply:
그렇군요! 자세한 설명 정말 감사합니다👍 항상 이유 있는 코드를 짜려고 노력하시는 모습 을 보며 많이 배워갑니다.